debian/patches/dsa/check_running_kernel_zstd_ubuntu_fix

```diff
@@ -0,0 +1,23 @@
+Index: pkg-nagios-plugins-contrib/dsa/checks/dsa-check-running-kernel
+===================================================================
+--- dsa-check-running-kernel.dist	2022-03-12 21:36:28.000000000 +0100
++++ dsa-check-running-kernel	2022-03-12 21:36:32.000000000 +0100
+@@ -162,6 +162,8 @@
+ 	cat_vmlinux "$image" "\x02\x21\x4c\x18"  "lz4 -dc"  0
+ 	# lzo compressed image
+ 	cat_vmlinux "$image" "\x89\x4c\x5a\x4f\x00\x0d\x0a\x1a"  "lzop -dc"  0
++	# zstd compressed image
++	cat_vmlinux "$image" "\x28\xb5"              "zstd -d" 0
+ 
+ 	echo "ERROR: Unable to extract kernel image." 2>&1
+ 	exit 1
+@@ -226,6 +228,9 @@
+ 	exit $WARNING
+ fi
+ 
++# Adjustment for PVE/Proxmox and Ubuntu kernels: Strip off the last fragment.
++on_disk_version=$(echo "$on_disk_version" | sed -E 's/(.*) \(.*\)$/\1/')
```

Comment:
### Introduction
In this comment, we wanted to share the outcomes from our findings on this matter, as a reasoning for this change.

> We discovered there was a suffix added to the on-disk kernel image, which we stripped off using sed in order to satisfy the comparison operation.

### Observations
```
strings 5erkernel.decompressed | grep "Linux version"
Linux version 5.3.18-3-pve (build@pve) (gcc version 8.3.0 (Debian 8.3.0-6)) #1 SMP PVE 5.3.18-3 (Tue, 17 Mar 2020 16:33:19 +0100) ()
```

```
Linux version 5.13.0-27-generic (buildd@lgw01-amd64-045) (gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #29~20.04.1-Ubuntu SMP Fri Jan 14 00:32:30 UTC 2022 (Ubuntu 5.13.0-27.29~20.04.1-generic 5.13.19)
```

### Evaluation
It was exactly this last suffix, either an empty `()`, or `(Ubuntu 5.13.0-27.29~20.04.1-generic 5.13.19)`, which caused an anomaly when comparing the on-disk with the running kernel, in turn croaking like

```
/usr/lib/nagios/plugins/check_running_kernel_ng
WARNING: Running kernel does not match on-disk kernel image: [Linux version 5.4.162-1-pve (build@proxmox) (gcc version 8.3.0 (Debian 8.3.0-6)) #1 SMP PVE 5.4.162-2 (Thu, 20 Jan 2022 16:38:53 +0100) != Linux version 5.4.162-1-pve (build@proxmox) (gcc version 8.3.0 (Debian 8.3.0-6)) #1 SMP PVE 5.4.162-2 (Thu, 20 Jan 2022 16:38:53 +0100) ()]
```

### Thoughts
We've only been able to spot this on non-vanilla Debian kernels. On a standard Debian bullseye machine running `5.10.0-12`, this is not an issue.

### Solution
So, we decided to strip off this suffix using `sed`. Can you think of any improvement on this matter?

Reply:
I've diverted this topic to #92 in order to keep things separate.